src/app/query/ProductList.query.js

```diff
@@ -312,9 +322,37 @@ export class ProductListQuery {
         ];
     }
 
+    /**
+     * Returns a field querying video-specific data for a media gallery entry.
+     * @returns {Field} the video_content field
+     * @private
+     */
+    _getVideoContentField() {
+        return new Field('video_content').addFieldList([
+            'media_type',
+            'video_description',
+            'video_metadata',
+            'video_provider',
+            'video_title',
+            'video_url'
+        ]);
+    }
+
+    /**
+     * Returns a field querying the thumbnail of a media gallery entry.
+     * Not to be confused with the product thumbnail field, which has the same name but is a direct subfield of the product
+     * @returns {Field}
+     * @private
+     */
+    _getMediaThumbnailField() {
+        return new Field('thumbnail').addField('url');
+    }
+
     _getMediaGalleryField() {
         return new Field('media_gallery_entries')
-            .addFieldList(this._getMediaGalleryFields());
+            .addFieldList(this._getMediaGalleryFields())
+            .addField(this._getVideoContentField())
```

Comment:
Please use only `addFieldList` and separate every field creating into a function

Reply:
Reference productList query for example